File: finder.c
#include <stdio.h>

int main()
{
  int i, k, a, b;
  int minnum;
  int maxnum;
  float n;
  int found = 0;

  printf("Give the key of ");
  scanf("%d", &i);
  if (i<2) {return 0;}

  printf("Search from where to where? (integer)\n");
  scanf("%d", &minnum);
  printf("to\n");
  scanf("%d", &maxnum);

  for (k=minnum; k<=maxnum; k++)
  {
    a=i-k;
    b=i-10*k;
    n=(double)a/(double)b;
      
    printf("When k=%d, a=%d, b=%d, so n=%.3f\n", k, a, b, n);
      
    if (n!=0 && n!=1 && n!=-1 && n/1.00 == (int)n)
    {
      if (n>i/2) {n-=i;}
      printf("\n★ For k=%d, %d's key is %d :)", k, i, (int)n);
      found=1;
      break;
    }
    /* else
    {
      printf("\n");
    } */
  }
  if (found!=1)
  {
    printf("None found in %d to %d :(", minnum, maxnum);
  }
  return 0;
}

File: finder.py
found = 0

i = int(input("Give the key of "))
if (i<2): sys.exit

minnum = int(input("Search from where to where? (integer)\n"))
maxnum = int(input("to\n"))

for k in range(minnum, maxnum+1, 1):
  a=i-k
  b=i-10*k
  n=round(a/b, 3)
  print ('When k =', k, ', a =', a, ', b =', b, ', so n =', n)

  if (n!=0 and n!=1 and n!=-1 and n%1==0):
    if (n>i/2): n-=i
    print('\n★ For k =', k, ',', i, '\'s key is', int(n), ':)')
    found = 1 
    break
    sys.exit
  #else:
    #print('\n')
if (found!=1):
  printf("None found in %d to %d :(", minnum, maxnum)
  sys.exit
